### PR TITLE
Initial release for support new formats db

### PR DESCRIPTION
**Pull request by Balferian**
**Item db module:**
- [x] added support for subtype.
- [x] adeed support for trade restrictions.
- [x] rewrited Equip(Locations/Upper/Jobs) functions.
- [x] fixed mobs drop list.

**Mob db module:**
- [x] added support for new AI parametr.
- [x] rewrited (element/race/size) functions.
- [x] fixed items drop list.

**Other:**
- [x] added support for mobs drop flag "nosteal".
- [x] added support for 4th jobs (config/jobs.php).
- [x] added support for random options (Renewal).
- [x] fixed items in cash shop module.
- [x] fixed items in character module.
- [x] fixed items in login view module.
- [x] fixed items in guild module.
- [x] fixed items in item shop module.
- [x] fixed items in logdata module.
- [x] fixed items in purchase module.
- [x] fixed items in vending module.
- [x] fixed items in buyingstore module.
- [x] fixed temporary table load (Thanks for @Lemongrass3110!).

p.s.: I rewrote this code in a couple of days.
If there are suggestions how to rewrite code better, I will be glad for help.

### CODE REVIEW

**Comment by reunite-ro** [comment]
Ive tried and yes, this PR makes the viewing character works again.
ItemDB and MobDB also works. Ill try it thoroughly later on.

I dont know whether this PR will be merged in master or not, but if yes :

1. I dont know if guild emblems should be merge with this PR or just for new formats db. Seems like this PR will be merged ? https://github.com/rathena/rathena/pull/5731.

2. And since you already add 4th jobs inside, maybe some of the remaining jobs which missed last update , Star Emperor and Soul Reaper in this PR https://github.com/rathena/FluxCP/pull/258 and also castle for WoE:TE https://github.com/rathena/FluxCP/pull/259.

Btw great works Balferian !

File: lib/Flux.php
Comment (sanasol) [inline lib/Flux.php:677]: I think we should check that array key exists, at least to avoid php warnings.

File: lib/Flux.php
Comment (sanasol) [inline lib/Flux.php:903]: Same here, check that key exists in array

File: lib/Flux.php
Comment (sanasol) [inline lib/Flux.php:912]: And here, check that key exists in array

File: modules/guild/view.php
Comment (sanasol) [inline modules/guild/view.php:156]: Typo

File: modules/item/index.php
Comment (sanasol) [inline modules/item/index.php:121]: Should we remove it or what?

File: modules/item/view.php
Comment (sanasol) [inline modules/item/view.php:27]: Looks weird, maybe we can do  this in some other way?

File: themes/default/monster/index.php
Comment (sanasol) [inline themes/default/monster/index.php:80]: Useless to keep two columns with same value, isn't it?

File: modules/item/index.php
Comment (Balferian) [reply modules/item/index.php:121]: combination search does not work now.
it needs to be rewritten or deleted.

File: themes/default/monster/index.php
Comment (sanasol) [reply themes/default/monster/index.php:80]: Maybe it should be name_japanese instead?

File: themes/default/monster/view.php
Comment (sanasol) [inline themes/default/monster/view.php:5]: I dont get it, everywhere replaced name_japanese to name_english, but here you replace iro_name with name_japanese.


Shouldn't be iro_name => name_english and kro => name_japanese?

**Review by sanasol — CHANGES_REQUESTED** [review]

File: themes/default/monster/view.php
Comment (sanasol) [reply themes/default/monster/view.php:5]: I mean whole `themes/default/monster/view.php` file, not only these lines.

**Comment by sanasol** [comment]
Reviewed code, but not tested, will try to do it during week.

Nice job.

File: modules/item/view.php
Comment (Balferian) [reply modules/item/view.php:27]: This is a draft of the code.
Maybe better to split the structure of jobs and classes (in config folder) into subarrays and get it depending on the type of server mechanics like:
```
'prerenewal' => array( list of possible jobs ),
'renewal' => array( list of possible jobs )
```
but I have not yet figured out how to do it better

Comment (sanasol) [reply modules/item/view.php:27]: Maybe we can split it like this, but merge it somehow while loading jobs list from file.

Like new function in `lib/Flux/Flux.php`
```php
public static function getEquipJobs()
{
	$EquipJobs = Flux::config('EquipJobs')->toArray();
         if($isRenewal) { ... }
	return $EquipJobs;
}
```

**Comment by admkakaroto** [comment]
@Balferian man, can you add this here too?
https://github.com/rathena/FluxCP/issues/266

Now its possible, using this PR:
https://github.com/rathena/rathena/pull/5731

**Comment by Akkarinage** [comment]
Summoning @Balferian - do you wanna finish this off an I'll merge it? 🥇

**Comment by RagnaWay** [comment]
Missing 4th jobs baby in jobs.php

**Comment by ecdarreola** [comment]
There seems to have a problem using latest rep I cannot view subtypes on item
image

**Comment by Rytech2** [comment]
In the equip_upper.php file add this line to recognize equips set for 4th jobs (will be a thing soon)....
`	'class_forth' => 'Forth'`
and in jobs_gender_linked.php file add these lines as the Kagerou/Oboro are sex split jobs....
`	4211 => 'Kagerou',
	4212 => 'Oboro',`
Everything else looks good.

**Comment by Rytech2** [comment]
I almost forgot about the existence of baby jobs. jobs_gender_linked.php also needs these added....
`	4223 => 'Baby Kagerou',
	4224 => 'Baby Oboro',`
Sorry about that. xD

**Comment by tony891254** [comment]
excuse me, cannot read the card inside the item,  item inside character like this


but in fluxcp card cannot read

**Comment by Balferian** [comment]
@tony891254 thanks!
The problem was fixed.

**Comment by tony891254** [comment]
> @tony891254 thanks!
> The problem was fixed.

thanks it works but in cart inventory not read yet

**Comment by ecdarreola** [comment]
Unable to add items in item_shop

image

**Comment by Balferian** [comment]
@tony891254 text and card inside the item from cart was fixed.

**Comment by ecdarreola** [comment]
MVP Ranking doesn't show or filter monster name or MVP name.

**Comment by Balferian** [comment]
> MVP Ranking doesn't show or filter monster name or MVP name.
fixed

**Comment by zdurexz** [comment]
issue : The problem found is not showing.
image

The code now shows just like this.
image

Desired result
image

**Comment by ecdarreola** [comment]
module item->edit not working.

**Comment by Seavoltos** [comment]
Character doesn't show in account view or ranking lvl

**Comment by Akkarinage** [comment]
@Balferian I'm looking to merge this within the next week - can you have a look at the remaining reviews from Sanasol please.

The item pages no longer need to have the copy/edit features, since it's far easier to just use the yml2sql converter than it is to keep these functions updated with new bullcarp from Gravity. The edit feature for mobs can be dropped entirely, too. This may also force users to start using their rAthena software correctly.. maybe..?

**Comment by mazvi** [comment]
> @Balferian I'm looking to merge this within the next week - can you have a look at the remaining reviews from Sanasol please.
> 
> The item pages no longer need to have the copy/edit features, since it's far easier to just use the yml2sql converter than it is to keep these functions updated with new bullcarp from Gravity. The edit feature for mobs can be dropped entirely, too. This may also force users to start using their rAthena software correctly.. maybe..?

Now is next week

**Comment by lelouch22** [comment]
This shows up when i tried to view  "My Account" or "Item List"
```
Error: Flux_Error

Message: Failed to import/replace rows from table 'ragnarok.item_db2'

File: /home/asprikar/public_html/lib/Flux/TemporaryTable.php:229
```

The item_db2 im using was this
https://github.com/rathena/rathena/blob/master/sql-files/item_db2.sql

**Comment by Lemongrass3110** [comment]
> This shows up when i tried to view "My Account" or "Item List"
> 
> ```
> Error: Flux_Error
> 
> Message: Failed to import/replace rows from table 'ragnarok.item_db2'
> 
> File: /home/asprikar/public_html/lib/Flux/TemporaryTable.php:229
> ```
> 
> The item_db2 im using was this https://github.com/rathena/rathena/blob/master/sql-files/item_db2.sql

I gave @Balferian a fix for this one. I dont know if he pushed it already to this PR, but I think he has not.

**Comment by Balferian** [comment]
@Akkarinage, I think this pull can be merged if you have no suggestions.
Latest commits include:
- fix for temporary tables from @Lemongrass3110.
- updated functions from Sanasol requests.
- Add/Edit/Copy pages removed.

I have a lot of updates/fixes for Flux, but I`ll be make new pulls after this pr will merge.

**Comment by RoAdmin01** [comment]
@Akkarinage please merge this :)

**Comment by mazvi** [comment]
> @Balferian I'm looking to merge this within the next week - can you have a look at the remaining reviews from Sanasol please.
> 
> The item pages no longer need to have the copy/edit features, since it's far easier to just use the yml2sql converter than it is to keep these functions updated with new bullcarp from Gravity. The edit feature for mobs can be dropped entirely, too. This may also force users to start using their rAthena software correctly.. maybe..?

Now its not only next week, but now its next year :D

**Comment by Everade** [comment]
Ready for merge @Akkarinage

**Comment by Benq28021997** [comment]
> 

new year no marge^^ @Akkarinage pls merg this to make flux great agin

**Review by Akkarinage — APPROVED** [review]

**Pull request merged**
